### PR TITLE
Add input sanitization setup

### LINKED ISSUE #294: Yaml view should hide `managedFields` by default
**Is your feature request related to a problem? Please describe.**
A lot of scrolling in kdash is dedicated to getting around the big `managedFields` object in `metadata` (something `kubectl` hides by default unless you specify `--show-managed-fields`). It would be nice if the output we showed was more useful that way. Given that this is TUI user app, it's probably OK to hide this controller specific (and very verbose) field.

**Describe the solution you'd like**
Mutate output from the apiservers list/watch calls before showing it to the user.
Use `.meta_mut()` go get and clear `.managed_fields` or use [`ResourceExt`](https://docs.rs/kube/latest/kube/trait.ResourceExt.html#tymethod.managed_fields_mut) directly to do it:

```rust
use kube::{Resource, ResourceExt};
fn hide_managed_fields<K: Resource>(obj: &mut K) {
    obj.managed_fields_mut().clear();
}
```

### PR DESCRIPTION
**Pull request by clux**
This adds a `sanitize_obj` fn in `utils` that is supposed to be used with the raw input from `kube` and used to initialise `k8s_obj` for each model. This can only be sensibly done when we have ownership of the object (it's weird to do it mutably at every render), so we have modified the input pipeline to have ownership over the resource when constructing and doing the clearing of undesired properties then. In particular, we have:

- added `From<k8s_openapi::SomeType> for OurKubeType` and used where possible
- removed `KubeResouce::from_api` from the trait (replaced by `From` impl)
- made all `From` impls take ownership of the objects (rather than taking a ref)
- add `utils::sanitize_obj` fn that filters out `managedFields` from `metadata` of a resource
- use `utils::sanitize_obj` on the input of all the `From` impls

There was no reason not to take ownership AFAIKT. It only required moving to `IntoIterator` over `Iterator` and with that change we were able to use the new `sanitize_obj` fn during the `::from` calls.

One negative side-effect (that may be able to be cleaned up) is the requirement to tell what type we are using `From` from... I.e. we need to import all the k8s_openapi types into the scope of `kube_api.rs`.

Closes #294

### CODE REVIEW

File: src/app/cronjobs.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use k8s_openapi::{api::batch::v1beta1::CronJob, chrono::Utc};
+use k8s_openapi::{api::batch::v1::CronJob, chrono::Utc};
```
Comment (clux) [inline src/app/cronjobs.rs:1]: note that [CronJob is stable since 1.21](https://kubernetes.io/blog/2021/04/09/kubernetes-release-1.21-cronjob-ga/)

Comment (deepu105) [reply src/app/cronjobs.rs:1]: Thanks for the catch

**Review by deepu105 — APPROVED** [review]
Thank you. This definitely is way better than what I had 🙂

**Comment by clux** [comment]
Think I've fixed what clippy/fmt and tests complained about, but didn't get tests to run locally out of the box (some problems with husky exiting) so hopefully the small fixes worked.

**Comment by clux** [comment]
ah, i guess we need to remove all the managedField data from all the tests as well

**Comment by clux** [comment]
..and they needed to be nulled out as well. sorry, hard to tell for sure what's going on when the local test setup is crashing locally.

**Comment by deepu105** [comment]
No worries and thank you for the PR :pray:

**Pull request merged**